### PR TITLE
build: Decouple and minimise dependencies for visualisation

### LINKED ISSUE #53: Reconsider tex2pix dependency
`tex2pix` is a [core dependency of `pylhe`](https://github.com/scikit-hep/pylhe/blob/d9ce20436161ec174bde6555c7dd6e68f7f5e0ea/setup.cfg#L40-L43) but the [last release of `v0.3.1` was in 2016](https://pypi.org/project/tex2pix/0.3.1/#history), so it seems to be safely "unmaintained" at this point. The source code also doesn't seem to be publicly available on GitHub.

:wave: @agbuckley, as the author of `tex2pix` (thanks! :bow:), can you comment on if there is any plans to maintain `tex2pix` in the future? Or would you recommend that we look for an alternative that is more probable to be patched if there are issues that are found. I can imagine that you already have more than enough on your plate and being responsible for maintaining another codebase might very understandably not be high on the list. :)

### PR DESCRIPTION
**Pull request by eduardo-rodrigues**
It is best to decouple as much as possible, in design, and with the dependency on Graphviz we now have for visualisation, we no longer need networkx, tex2pix and dot2tex. The package gets much lighter with no loss of functionality since, as I show in MR https://github.com/scikit-hep/pylhe/pull/130, it is trivial at present to visualize a graph directly in a notebook or simply save the graph in, say, PNG or PDF.

IMO this deals with https://github.com/scikit-hep/pylhe/issues/53.

Resolves #53

```
* Remove visualize method.
* Remove viz extra.
* Remove networkx and tex2pix dependencies in favor of using graphviz.
```

### CODE REVIEW

File: tests/test_visualize.py [hunk 1]
```diff
@@ -1,21 +1,8 @@
...
 
-def test_visualize(tmpdir):
-    lhe_file = skhep_testdata.data_path("pylhe-testfile-pr29.lhe")
-    events = pylhe.read_lhe_with_attributes(lhe_file)
-
-    start_event = 1
-    stop_event = 2
-    filename = tmpdir.join(f"event{start_event}.pdf")
-    for event in itertools.islice(events, start_event, stop_event):
-        pylhe.visualize(event, filename)
```
Comment (matthewfeickert) [inline tests/test_visualize.py:16]: Any objection to keeping this in to make sure that the graphs being produced are actually valid for `graph.render` but with

```python
event = next(itertools.islice(events, 1, 2))
event.graph.render(filename="test_event1", format="pdf", cleanup=True)
```

?

**Review by matthewfeickert — COMMENTED** [review]
I think this is good, and solves a lot of problems. :rocket: I think the only thing that would be good to note/fix(?) for the future would be

that the old method allowed for LaTeX fonts


old_example

while using `graphviz.graphs.Digraph.render` doesn't seem to do that

test

Is there a way to get LaTeX fonts? This is somewhat steering back into the questions of Issue #53, which I haven't taken the time to review and refresh myself on, so apologies if all of this should be obvious.

File: tests/test_visualize.py (hunk 1 above)
Comment (eduardo-rodrigues) [reply tests/test_visualize.py:16]: That sounds good. You're basically suggesting to test what is exemplified in the README, and that only fair.

True that test coverage is not amazing yet and needs more work.

**Comment by eduardo-rodrigues** [comment]
> I think the only thing that would be good to note/fix(?) for the future would be that the old method allowed for LaTeX fonts.

> Is there a way to get LaTeX fonts? This is somewhat steering back into the questions of Issue https://github.com/scikit-hep/pylhe/issues/53, which I haven't taken the time to review and refresh myself on, so apologies if all of this should be obvious.

It must be possible since both packages in the end write DOT language files. I used what I'm doing in DecayLanguage. I'm putting your request down as a to-do to look into this soon, after a few days off ... In the meantime, if this goes in we're good tp proceed with the other MRs dependent on this.

File: tests/test_visualize.py (hunk 1 above)
Comment (eduardo-rodrigues) [reply tests/test_visualize.py:16]: Now added via https://github.com/scikit-hep/pylhe/pull/131/commits/2c37af784231d618f05b0fe60d2a3909e65b5e0c.

**Comment by eduardo-rodrigues** [comment]
Hello @matthewfeickert, I finally remembered to re-instate a test as you had before, see your comment https://github.com/scikit-hep/pylhe/pull/131#discussion_r901226407 above.

Coverage is to be worked on but that's not specific to this MR, rather a general issue that coverage is missing for certain classes, etc. That's clearly outsidethe scope of this MR. This is an important MR as it makes dependencies easier and unblocks some dependent MRs.

As for your comment https://github.com/scikit-hep/pylhe/pull/131#issuecomment-1162397488 I would gladly make this a follow-up issue as it is a quality of life improvement, and something I can look into in the future. Again, a different matter, second priority.

In short, I really think this is ready to do. If happy please approve and I will make the follow-up issue. Thanks a lot.

**Comment by eduardo-rodrigues** [comment]
Follow-up issue created - https://github.com/scikit-hep/pylhe/issues/135 for discussion https://github.com/scikit-hep/pylhe/pull/131#issuecomment-1162397488.

**Comment by eduardo-rodrigues** [comment]
Hi @matthewfeickert, I will merge this really shortly since the follow-up task is created and blocking this further is counter-productive and not helping anyone, certainly not the users. Coverage is a serious matter to deal with across the files, not just the 2-3 lines mentioned by codecov above ;-).

**Comment by matthewfeickert** [comment]
> Hi @matthewfeickert, I will merge this really shortly since the follow-up task is created and blocking this further is counter-productive and not helping anyone, certainly not the users. Coverage is a serious matter to deal with across the files, not just the 2-3 lines mentioned by codecov above ;-).

Okay. :+1: I still have some concerns about trading one viz library for another that now changes how the visualizations look, but I'm willing to kick that down the road a bit to sometime when I have more time to do the work here for these concerns to matter.
